Sources/eppo/dto/EppoValue.swift

```diff
@@ -2,154 +2,167 @@ import Foundation
     }
-    
+
     public static func valueOf(_ value: Double) -> EppoValue {
-        return EppoValue(value: String(value), type: EppoValueType.Number);
+        return EppoValue(value: value);
     }
     
+    public static func valueOf(_ value: Int) -> EppoValue {
+        return EppoValue(value: value);
+    }
+
     public static func valueOf(_ value: String) -> EppoValue {
-        return EppoValue(value: value, type: EppoValueType.String);
+        return EppoValue(value: value);
     }
 
     public static func valueOf(_ value: [String]) -> EppoValue {
         return EppoValue(array: value);
     }
+    
+    public func isNull() -> Bool {
+        return self.type == EppoValueType.Null
+    }
+    
+    public func isBool() -> Bool {
+        return self.type == EppoValueType.Boolean
+    }
+    
+    public func isNumeric() -> Bool {
+        return self.type == EppoValueType.Numeric
+    }
 
-    public func boolValue() throws -> Bool {
-        if self.value == nil {
-            throw Errors.valueNotSet;
-        }
-
-        guard let rval = Bool(self.value!) else {
-            throw Errors.conversionError;
-        }
-
-        return rval;
+    public func isString() -> Bool {
+        return self.type == EppoValueType.String
     }
     
-    public func doubleValue() throws -> Double {
-        if self.value == nil {
-            throw Errors.valueNotSet;
+    public func getBoolValue() throws -> Bool {
+        guard let value = self.boolValue else {
+            throw Errors.valueNotSet
         }
+        return value
+    }
 
-        guard let rval = Double(self.value!) else {
-            throw Errors.conversionError;
+    public func getDoubleValue() throws -> Double {
+        guard let value = self.doubleValue else {
+            throw Errors.valueNotSet
         }
-
-        return rval;
+        return value
     }
 
-    public func arrayValue() throws -> [String] {
-        if self.array == nil {
-            throw Errors.valueNotSet;
+    public func getStringArrayValue() throws -> [String] {
+        guard let value = self.stringArrayValue else {
+            throw Errors.valueNotSet
         }
 
-        return self.array!;
+        return value;
     }
 
-    public func stringValue() throws -> String {
-        if self.value == nil {
-            throw Errors.valueNotSet;
+    public func getStringValue() throws -> String {
+        guard let value = self.stringValue else {
+            throw Errors.valueNotSet
         }
 
-        return self.value!;
+        return value;
     }
 
     public func toHashedString() -> String {
+        // todo: this is a bit of a hack. we should probably just use a switch
+        // to handle the different types.
         var str = ""
-        if let value = self.value {
+        if let value = self.stringValue {
             str = value
-        } else if let array = self.array {
+        } else if let array = self.stringArrayValue {
```

Comment:
I don't know we've aligned on a standard but in Java/Android we have ea space before the comma in our join `" ,"` for string value of array of strings